### PR TITLE
Run simulations concurrently.

### PR DESCRIPTION
**Pull request by mwhittaker**
This PR changes the `Simulator.Run` method to run multiple simulations in parallel. This does speed things up a bit (5x throughput for TestPassingSimulations), but things still aren't as fast as I'd like. I also need a lot of parallelism to speed things up fully which I don't yet fully understand. I'll try to do some profiling and optimizations in future PRs.

### CODE REVIEW

File: internal/sim/api.go [hunk 1]
```diff
@@ -313,50 +315,81 @@ func validateWorkload(t reflect.Type) error {
...
-	// TODO(mwhittaker): Run simulations in multiple goroutines.
-	// TODO(mwhittaker): Read and run graveyard entries.
+	// Spawn n concurrent simulators. The simulators read options from the opts
+	// channel and write errors and results to the errs and results channels.
+	//
+	// TODO(mwhittaker): Optimize things and pick a smarter value of n.
+	n := 1000
+	opts := make(chan options, n)
+	errs := make(chan error, n)
+	results := make(chan Results, n)
```
Comment (spetrovic77) [inline internal/sim/api.go:332]: Rename errs and results to describe what type of errors they store.

File: internal/sim/api.go [hunk 2]
```diff
@@ -313,50 +315,81 @@ func validateWorkload(t reflect.Type) error {
...
 						NumOps:      numOps,
 						NumReplicas: 1,
 						FailureRate: failureRate,
 						YieldRate:   yieldRate,
 					}
-					results, err := s.runOne(ctx, opts)
-					if err != nil && err == ctx.Err() {
+
+					select {
+					case <-ctx.Done():
```
Comment (spetrovic77) [inline internal/sim/api.go:378]: This processing is confusing, perhaps better to run it in a separate goroutine?

File: internal/sim/api.go [hunk 3]
```diff
@@ -313,50 +315,81 @@ func validateWorkload(t reflect.Type) error {
...
 	defer cancel()
 
-	// TODO(mwhittaker): Use a smarter algorithm to sweep over hyperparameters.
-	// TODO(mwhittaker): Run simulations in multiple goroutines.
-	// TODO(mwhittaker): Read and run graveyard entries.
+	// Spawn n concurrent simulators. The simulators read options from the opts
+	// channel and write errors and results to the errs and results channels.
+	//
+	// TODO(mwhittaker): Optimize things and pick a smarter value of n.
+	n := 1000
```
[the remark this answers is not in the document]
Comment (mwhittaker) [reply internal/sim/api.go]: Done.

File: internal/sim/api.go (hunk 1 above)
Comment (mwhittaker) [reply internal/sim/api.go]: Done. I also added some more comments.

File: internal/sim/api.go (hunk 2 above)
Comment (mwhittaker) [reply internal/sim/api.go]: Good idea. Done.

**Review by mwhittaker — COMMENTED** [review]
Thanks Srdjan!

**Review by spetrovic77 — APPROVED** [review]
Looks great.

**Pull request merged**